integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/APIKeyHeaderTestCase.java

```diff
@@ -1,116 +0,0 @@
-
-import com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpStatus;
-import org.json.JSONObject;
-import org.testng.Assert;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
-import org.wso2.am.integration.clients.publisher.api.ApiResponse;
-import org.wso2.am.integration.clients.publisher.api.v1.dto.APIOperationsDTO;
-import org.wso2.am.integration.clients.store.api.v1.dto.APIKeyDTO;
-import org.wso2.am.integration.test.utils.bean.APIRequest;
-import org.wso2.choreo.connect.tests.apim.ApimBaseTest;
-import org.wso2.choreo.connect.tests.apim.dto.AppWithConsumerKey;
-import org.wso2.choreo.connect.tests.apim.dto.Application;
-import org.wso2.choreo.connect.tests.apim.utils.PublisherUtils;
-import org.wso2.choreo.connect.tests.apim.utils.StoreUtils;
-import org.wso2.choreo.connect.tests.util.*;
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-public class APIKeyHeaderTestCase extends ApimBaseTest {
-
-    private static final String SAMPLE_API_NAME = "APIKeyHeaderTestAPI";
-    private static final String SAMPLE_API_CONTEXT = "apiKeyHeader";
-    private static final String SAMPLE_API_VERSION = "1.0.0";
-    private static final String APP_NAME = "APIKeyHeaderTestApp";
-
-    protected String apiKey;
-    private String applicationId;
-    private String apiId;
-    private String endPoint;
-    private String internalKey;
-
-    @BeforeClass(description = "Initialise the setup for API key tests")
-    void start() throws Exception {
-        super.initWithSuperTenant();
-
-        String targetDir = Utils.getTargetDirPath();
-        String filePath = targetDir + ApictlUtils.OPENAPIS_PATH + "api_key_openAPI.yaml";
-
-        JSONObject apiProperties = new JSONObject();
-        apiProperties.put("name", SAMPLE_API_NAME);
-        apiProperties.put("context", "/" + SAMPLE_API_CONTEXT);
-        apiProperties.put("version", SAMPLE_API_VERSION);
-        apiProperties.put("provider", user.getUserName());
-        apiId = PublisherUtils.createAPIUsingOAS(apiProperties, filePath, publisherRestClient);
-
-        publisherRestClient.changeAPILifeCycleStatus(apiId, "Publish");
-
-        //Create and subscribe to app
-        Application app = new Application(APP_NAME, TestConstant.APPLICATION_TIER.UNLIMITED);
-        applicationId = StoreUtils.createApplication(app, storeRestClient);
-
-        PublisherUtils.createAPIRevisionAndDeploy(apiId, publisherRestClient);
-
-        StoreUtils.subscribeToAPI(apiId, applicationId, TestConstant.SUBSCRIPTION_TIER.UNLIMITED, storeRestClient);
-
-        endPoint = Utils.getServiceURLHttps(SAMPLE_API_CONTEXT + "/1.0.0/pet/1");
-
-        // Obtain internal key
-        ApiResponse<org.wso2.am.integration.clients.publisher.api.v1.dto.APIKeyDTO> internalApiKeyDTO =
-                publisherRestClient.generateInternalApiKey(apiId);
-        internalKey = internalApiKeyDTO.getData().getApikey();
```

Comment:
This test case tests both internal key and apikey for the same API. Shall we keep this?

Reply:
+1. There was a use case to have both internal-key and API key the same name.

Reply:
There were two cases. One had both APIkey and Internal key with the same name and both invoked the same API. Then there was another case where the APIkey and internal key had there original forms but invoke the same API. I think we are missing the second case.

Reply:
Added all existing testcases. The second case it also covered.